### PR TITLE
1069: Customer info details serialize/deserialize

### PR DESCRIPTION
**Pull request by jorgesanchezperez**
To parse properly the json value birthdate with format "dd/MM/yyyy" it's necessary implement the proper deserialiser to parse the text to a LocalDate object that represents the birthDate of the customer.
- Added new serialiser and deserialiser for LocalDate joda object.
    
Issue: https://github.com/SecureApiGateway/SecureApiGateway/issues/1069

### CODE REVIEW

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import org.joda.time.LocalDate;
+import org.joda.time.format.DateTimeFormat;
+
+import java.io.IOException;
+
+/**
+ * Common deserializer to parse a json value with format "dd/MM/yyyy" to {@link LocalDate}
+ */
+public class LocalDateTimeDeserializer  extends StdDeserializer<LocalDate> {
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeDeserializer.java:14]: Rename to LocalDateDeserializer

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    public LocalDateTimeDeserializer(Class<?> vc) {
+        super(vc);
+    }
+
+    @Override
+    public LocalDate deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
+            throws IOException {
+        String date = jsonParser.getText();
+        return LocalDate.parse(date, DateTimeFormat.forPattern(LOCAL_DATE_FORMAT));
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeDeserializer.java:29]: store DateTimeFormat.forPattern(LOCAL_DATE_FORMAT) and reuse it

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeSerializer.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public LocalDateTimeSerializer(Class<LocalDate> t) {
+        super(t);
+    }
+
+    @Override
+    public void serialize(LocalDate localDate, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+            throws IOException {
+        jsonGenerator.writeObject(DateTimeFormat.forPattern(LOCAL_DATE_FORMAT).print(localDate));
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeSerializer.java:29]: Use jsonGenerator.writeString method

**Review by dbadham-fr — CHANGES_REQUESTED** [review]

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeDeserializer.java (hunk 2 above)
Comment (jorgesanchezperez) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeDeserializer.java]: I have created an interface to store it and reuse it in the serialiser and deserialiser.

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeSerializer.java (hunk 3 above)
Comment (jorgesanchezperez) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/jackson/LocalDateTimeSerializer.java]: done!

**Review by dbadham-fr — APPROVED** [review]

**Pull request merged**